.github/scripts/test_print_latest_commits.py

```diff
@@ -16,33 +32,48 @@ def make_test_checks(self) -> List[Dict[str, Any]]:
 
 class TestPrintCommits(TestCase):
     @mock.patch('print_latest_commits.get_commit_results', return_value=TestChecks().make_test_checks())
-    def test_match_rules(self, mock_get_commit_results: Any) -> None:
-        "Test that passes all conditions for promote-able"
+    def test_all_successful(self, mock_get_commit_results: Any) -> None:
+        "Test with workflows are successful"
+        workflow_checks = mock_get_commit_results()
+        self.assertTrue(isGreen("sha", workflow_checks))
+
+    @mock.patch('print_latest_commits.get_commit_results', return_value=TestChecks().make_test_checks())
+    def test_necessary_successful(self, mock_get_commit_results: Any) -> None:
+        "Test with necessary workflows are successful"
         workflow_checks = mock_get_commit_results()
+        for i in range(8, 13):
+            workflow_checks[i]['conclusion'] = 'failed'
         self.assertTrue(isGreen("sha", workflow_checks))
 
     @mock.patch('print_latest_commits.get_commit_results', return_value=TestChecks().make_test_checks())
-    def test_jobs_failing(self, mock_get_commit_results: Any) -> None:
-        "Test with one job failing, no pending jobs, at least 20 jobs run"
+    def test_necessary_skipped(self, mock_get_commit_results: Any) -> None:
+        "Test with necessary job (ex: pull) skipped"
         workflow_checks = mock_get_commit_results()
-        workflow_checks[0]['conclusion'] = 'failed'
-        # self.assertFalse(isGreen(workflow_checks))
+        workflow_checks[0]['conclusion'] = 'skipped'
+        self.assertEqual(isGreen("sha", workflow_checks), "pull checks were not successful")
 
     @mock.patch('print_latest_commits.get_commit_results', return_value=TestChecks().make_test_checks())
-    def test_pending_jobs(self, mock_get_commit_results: Any) -> None:
-        "Test with pending jobs, all jobs passing, at least 20 jobs run"
+    def test_skippable_skipped(self, mock_get_commit_results: Any) -> None:
+        "Test with skippable jobs (periodic and docker-release-builds skipped"
         workflow_checks = mock_get_commit_results()
-        workflow_checks[0]['conclusion'] = 'pending'
-        # self.assertFalse(isGreen(workflow_checks))
+        workflow_checks[6]['conclusion'] = 'skipped'
+        workflow_checks[7]['conclusion'] = 'skipped'
+        self.assertTrue(isGreen("sha", workflow_checks))
```

Comment:
Wait so does isGreen return a boolean OR a string? This isn't a blocker to land these tests but that doesn't seem very clear

Reply:
yeah, it returns a boolean True if the test is promoteable, but a string if it's not promoteable